src/Mod/Sketcher/Gui/CommandSketcherBSpline.cpp

```diff
@@ -1055,6 +1055,8 @@ void CmdSketcherInsertKnot::activated(int iMsg)
                                 .arg(QString::fromUtf8(multList.str().c_str())));
         // use an appropriate middle value from `knots`
         knotDialog.setDoubleValue(0.5 * (knots.front() + knots.back()));
+        // set step size dependent on the knot range
+        knotDialog.setDoubleStep(0.001 * (knots.back() - knots.front()));
         // use min/max from `knots`
         knotDialog.setDoubleRange(knots.front(), knots.back());
```

Comment:
It seems you must explicitly set the Min and Max value. Because as it is, I can enter e.g. "3000". The spin edit should only allow to enter geometrically valid values.

Reply:
I believed that was handled by `setDoubleRange`. I'll change it if needed.

Reply:
I can confirm you cannot enter values outside the range. If you press a key that would make the number invalid or have more decimal points, the value doesn't change. Same happens if you paste an invalid value.

Reply:
Does not work for me, here is a screencast:

I can enter any value there into the dialog but I should only be able to enter sensible values

Reply:
As such I am not particularly happy about the restricted decimal places but this appears to be a Qt issue: either we can't enter a value with more decimal places or all the unwanted decimal places are shown at all times.

Reply:
Hmm, GitHub seems to have problems with images. However, just pen the dialog and enter there some random value
-> result you can do this. If it would be correct and e.g. the range is between 0 and 1 you would not be able to enter a value like "4362"

Reply:
> Does not work for me, here is a screencast:

I can't see the screencast. But if that's the case it might be due to difference in Qt versions. Can you try replacing `knotDialog.setDoubleRange(knots.front(), knots.back());` with `setDoubleMaximum/Minimum` in your branch?

Reply:
> As such I am not particularly happy about the restricted decimal places but this appears to be a Qt issue: either we can't enter a value with more decimal places or all the unwanted decimal places are shown at all times.

I don't understand. When the user decided to have 2 decimals, it is his decision. When he realizes, he needs more, he can change this. For example I use 3 digits.

Reply:
> it might be due to difference in Qt versions. Can you try replacing `knotDialog.setDoubleRange(knots.front(), knots.back());` with `setDoubleMaximum/Minimum` in your branch?

I cannot compile until next week since I am traveling. Here is my FC version:
OS: Windows 10 Version 2009
Word size of FreeCAD: 64-bit
Version: 0.20.27422 (Git)
Build type: Release
Branch: master
Hash: d938733eaf2c2ce7cb18d1cbb56147185c473530
Python version: 3.8.6+
Qt version: 5.15.2
Coin version: 4.0.1
OCC version: 7.5.3
Locale: German/Germany (de_DE)

Reply:
> > As such I am not particularly happy about the restricted decimal places but this appears to be a Qt issue: either we can't enter a value with more decimal places or all the unwanted decimal places are shown at all times.
> 
> I don't understand. When the user decided to have 2 decimals, it is his decision. When he realizes, he needs more, he can change this. For example I use 3 digits.
The reported values and values used for calculations should be different, specifically the latter should store more digits. This can add unwanted rounding errors. For example say for whatever reason we have 
1. we have precision set to 1 digit (for exaggeration but we can conjure this issue with higher precision),
2. `v1 = 0.5`, `v2 = 0.6`, `vmid = (v1 + v2)/2`,
2. `vmid` is also stored in a user visible field.
Now, say we wanted `vmid**2`. Then, if `vmid` is stored precisely, the value (which is exactly `0.3025`) should be shown as `0.3`. But if `vmid` is stored incorrectly we are stuck with `vmid=0.6` and `vmid**2=0.36~0.4`.

In our case with a precision of 2 decimals a parameter change of 0.01 can make a lot of difference in the position of the point itself.

Additionally, we might want different precision for different types of values, like wanting length accurate to 0.01 mm but weight to 0.1 g.

I think I'm already dragging this discussion to a totally different tangent so if this is still relevant in the future I'll make a forum post/issue/PR.

Reply:
> > it might be due to difference in Qt versions. Can you try replacing `knotDialog.setDoubleRange(knots.front(), knots.back());` with `setDoubleMaximum/Minimum` in your branch?
> 
> I cannot compile until next week since I am traveling.

So I presume you are using the daily builds for testing?

I cannot replicate your issue with the following:
OS: Manjaro Linux (GNOME/gnome)
Word size of FreeCAD: 64-bit
Version: 0.20.27456 +1 (Git)
Build type: Release
Branch: sketcher-fix-6323
Hash: d23e445699cc94096116e4e7ce2512884d5fa098
Python version: 3.10.2
Qt version: 5.15.2
Coin version: 4.0.1
OCC version: 7.5.3
Locale: English/United States (en_US)

The differences then (in your version) are:
OS: Windows 10 Version 2009
Version: 0.20.27422 (Git)
Branch: master
Hash: d938733eaf2c2ce7cb18d1cbb56147185c473530
Python version: 3.8.6+
Locale: German/Germany (de_DE)

I don't have a windows build ready, bet I'll update once I've compiled with d938733eaf2c2ce7cb18d1cbb56147185c473530. I think the quickest thing you can try on your end is changing locale (if en_US is available)?

Reply:
Update: I can confirm arbitrary values can be inserted with d938733eaf2c2ce7cb18d1cbb56147185c473530. When I look closely I notice this is before #5517 was merged, so the permitted values are still [`-DBL_MAX`, 'DBL_MAX`]. The range should be fixed by 0.20.27456, but reopen this conversation and let me know if it isn't.